src/Analyzer/Pass/Expression/ArrayIllegalOffsetType.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * @author Kévin Gomez https://github.com/K-Phoen <contact@kevingomez.fr>
+ */
+
+namespace PHPSA\Analyzer\Pass\Expression;
+
+use PhpParser\Node\Expr;
+use PHPSA\Analyzer\Pass;
+use PHPSA\Context;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+
+class ArrayIllegalOffsetType implements Pass\AnalyzerPassInterface, Pass\ConfigurablePassInterface
+{
+    /**
+     * @param Expr\Array_|Expr\Assign $expr
+     * @param Context $context
+     * @return bool
+     */
+    public function pass(Expr $expr, Context $context)
+    {
+        if ($expr instanceof Expr\Array_) {
+            return $this->analyseArray($expr, $context);
+        } else if ($expr instanceof Expr\Assign && $expr->var instanceof Expr\ArrayDimFetch) {
+            return $this->analyseDimensionFetch($expr->var, $context);
+        }
+    }
+
+    /**
+     * @param Expr\ArrayDimFetch $expr
+     * @param Context $context
+     *
+     * @return bool
+     */
+    private function analyseDimensionFetch(Expr\ArrayDimFetch $expr, Context $context)
+    {
+        return $this->analyseExpression($expr->dim, $context);
+    }
+
+    /**
+     * @param Expr\Array_ $expr
+     * @param Context $context
+     *
+     * @return bool
+     */
+    private function analyseArray(Expr\Array_ $expr, Context $context)
+    {
+        $result = false;
+
+        /** @var Expr\ArrayItem $item */
+        foreach ($expr->items as $item) {
+            if ($item->key === null) {
+                continue;
+            }
+
+            $result = $this->analyseExpression($item->key, $context) || $result;
```

Comment:
`|| $result` is not needed since $result will always have the value of analyzeExpression at the end.